pkg/index/index.go

```diff
@@ -55,9 +59,13 @@ func New(rewriters []PathRewriter) *State {
 		shardClusterParentCluster: map[string]map[logicalcluster.Name]logicalcluster.Name{},
 		shardBaseURLs:             map[string]string{},
 		// Experimental feature: allow mounts to be used with Workspaces
-		// structure: (clusterName, workspace name) -> string serialized mount objects
+		// structure: (shard, logical cluster, workspace name) -> string serialized mount objects
 		// This should be simplified once we promote this to workspace structure.
-		clusterWorkspaceMountAnnotation: map[logicalcluster.Name]map[string]string{},
+		shardClusterWorkspaceMountAnnotation: map[string]map[logicalcluster.Name]map[string]string{},
```

Comment:
We have few duplicate ones here. We could do some more nicer abstraction here.
Most data is `shard` -> `cluster`-> `workspace` -> `data` - so some in-memory store mini-engine could help. Now its very hard to make changes where and make sure we clean, add, and update when we need to. 

Something to think of during insomnia nights.